### DISCUSSION TITLE
Computing mesh with toMesh and then determining normals

### DISCUSSION THREAD

**Discussion by andre-caldas** [gismo/gismo#730] [Q&A]
I am really, really new to G+Smo. I am sorry if I say something silly. :-)

The method `gsSurface<T>::toMesh()` creates a `gsGridIterator` for the surface domain to `addVertex` to the mesh. Then it computes faces.

If after calling `toMesh()`, I want to compute the normals for each vertex. Of course, I could simply reproduce the internals of `toMesh`, and construct the same `gsGridIterator` and compute the mesh at the iterated points. However, I do not feel that depending on the internals of `toMesh()` is a good thing.

I would like to know (or propose) an elegant solution to this problem. Maybe `toMesh()` could return a copy of the iterator...

What do you think? Am I missing anything?

Reference:
https://github.com/gismo/gismo/blob/a9917ca7994633250ed2ae5686169febd46d64c3/src/gsCore/gsSurface.hpp#L25

PS: It is interesting that we do not want to use the four parameter version of `addFace` inside the double loop.

**Comment by andre-caldas** [comment]
Of course, it would be more "generic" if we constructed and passed the iterator to `toMesh()` ourselves.

**Comment by hverhelst** [comment]
@filiatra would you have a suggestion here?
